src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java

```diff
@@ -0,0 +1,457 @@
+                                if (!street.getOsmId().equals(wayLabel)) {
+                                    LOG.warn("Mismatched path street edge with way ID {} along way {}", street.getOsmId(), wayLabel);
+                                    goodPath = false;
+                                } else {
+                                    matchedEdges.add(street);
+
+                                    // TODO: surely there's an easier way to find the back edge.
+                                    // add reverse of this edge
+                                    boolean foundReverse = false;
+                                    for (Edge reverse : wayEdges) {
+                                        if (reverse.isReverseOf(street)) {
+                                            foundReverse = true;
+                                            matchedEdges.add((StreetEdge) reverse);
+                                        }
+                                    }
+                                    if (!foundReverse) {
+                                        LOG.warn("No reverse found for matched SPT edge {}: {}", edge.getName(), ((StreetEdge) edge).getOsmId());
+                                    }
+                                }
+                            } else {
+                                LOG.warn("Edge on SPT is not a StreetEdge!  Cannot use it.");
+                            }
+                        }
+                    }
+
+                    options.cleanup();
+
+                    if (!goodPath) {
+                        LOG.warn("Bad path for way {}: {}", wayLabel, wayName);
+                        continue;
+                    }
+                }
+            }
+
+            // sanity check
+            LOG.info("Have {} matched street edges for way {}.", matchedEdges.size(), wayLabel + " " + wayName);
+            boolean isBad = false;
+            for (StreetEdge edge : matchedEdges) {
+                if (!edge.getOsmId().equals(wayLabel)) {
+                    LOG.warn("Found bad matched way!  Got: {} Should be: {}", edge.getOsmId(), wayLabel);
+                    isBad = true;
+                }
+            }
+
+            if (isBad) {
+                continue;
+            }
+
+            ///////////////////////////////////////////////////////////////////////
+            // If got this far, have a good set of matched way edges to work with.
+            // Now go set properties on them.
+            ///////////////////////////////////////////////////////////////////////
+
+            // TODO: Set edge properties
+        }
+
+        LOG.info("Found {} start and {} end intersections.", foundStartCt, foundEndCt);
+        if (missingStartCt > 0 || missingEndCt > 0) {
+            LOG.warn("Missing {} start intersections.", missingStartCt);
+            LOG.warn("Missing {} end intersections.", missingEndCt);
+        }
+    }
```

Comment:
The entire `build` function?  The intent is that it will eventually set properties on the existing graph edges identified using the shapefile segment's properties.  It does not do that yet.  Yes, more functions would be good.

The `build` function for the existing graph builders creates an entire graph from the loaded information, which is not what we want to do here; we just want to set additional information on a graph already built from data imported from OpenStreetMap.